agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -839,6 +841,56 @@ func TestAWSLogsDriverDatetimeFormat(t *testing.T) {
 	assert.Equal(t, *resp.Events[1].Message, "May 01, 2017 19:00:04 Agent\nRunning\nin the instance\n", fmt.Sprintf("Got log events message unexpected: %s", *resp.Events[1].Message))
 }
 
+// This tests the workflow for retriving private registry authentication data
+// from AWS Secrets Manager
+func TestPrivateRegistryAuthOverASM(t *testing.T) {
+
+	if os.Getenv("TEST_DISABLE_EXECUTION_ROLE") == "true" {
+		t.Skip("TEST_DISABLE_EXECUTION_ROLE was set to true")
+	}
+
+	RequireDockerVersion(t, ">=17.06.0-ce")
+
+	secretName := "FunctionalTest-PrivateRegistryAuth"
+	asmClient := secretsmanager.New(session.New(), aws.NewConfig().WithRegion(*ECS.Config.Region))
+	input := &secretsmanager.CreateSecretInput{
+		Description:  aws.String("Resource created for the ECS Agent Functional Test: TestPrivateRegistryAuthOverASM"),
+		Name:         aws.String(secretName),
+		SecretString: aws.String("{\"username\":\"user\",\"password\":\"swordfish\"}"),
+	}
+
+	// create secret value if it does not exist
+	_, err := asmClient.CreateSecret(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case secretsmanager.ErrCodeResourceExistsException:
+				t.Logf("AWS Secrets Manager resource already exists")
+				break
+			default:
+				require.NoError(t, err, "AWS Secrets Manager CreateSecret call failed")
+			}
+		}
+	}
+
+	agent := RunAgent(t, nil)
+	defer agent.Cleanup()
+
+	agent.RequireVersion(">=1.18.0")
+
+	tdOverrides := make(map[string]string)
+	tdOverrides["$$$REPOSITORY_CREDENTIALS$$$"] = secretName
+	tdOverrides["$$$EXECUTION_ROLE$$$"] = os.Getenv("ECS_FTS_EXECUTION_ROLE")
+
+	task, err := agent.StartTaskWithTaskDefinitionOverrides(t, "private-registry-auth-asm-validator-unix", tdOverrides)
+	require.NoError(t, err, "Expected to start task using private registry authentication over asm failed")
+
+	err = task.WaitStopped(2 * time.Minute)
```

Comment:
Can you use the constants defined [here](https://github.com/aws/amazon-ecs-agent/blob/master/agent/functional_tests/tests/functionaltests_test.go#L42)